### PR TITLE
Java Lab: Add NO_AUDIO signal to theater

### PR DESCRIPTION
**Pull request by molly-moen**
We were seeing issues where Safari would not process our [empty audio file](https://github.com/code-dot-org/javabuilder/blob/31a7806859a3b68a95ddd5c0addeebbf2bd77ae0/org-code-javabuilder/media/src/main/java/org/code/media/support/AudioWriter.java#L77) for Theater projects without audio, which meant if a Theater project did not have audio, it would never show the gif on Safari. Since we wanted to add a `NO_AUDIO` signal anyway, this PR adds a `NO_AUDIO` signal to the frontend and counts that as a signal that is ready for playback. Once this is deployed I will merge the Javabuilder-side PR that sends a `NO_AUDIO` signal instead of an empty audio file.

## Links

- jira ticket: [SL-314](https://codedotorg.atlassian.net/browse/SL-314)

## Testing story
In testing this I found that Safari also seems to have issues with how we create audio files locally, so testing on Safari against localhost Javabuilder for a theater project with audio does not work (it won't play because it doesn't get a `oncanplaythrough` event). However, I tested against a deployed version of Javabuilder and that worked fine.

I tested that now Safari, Firefox and Chrome will play a theater project with and without audio, against a dev deploy of Javabuilder with a new `NO_AUDIO` signal. They also will continue to act as expected against the current version of Javabuilder.

## Deployment strategy
This change must be deployed before the corresponding Javabuilder change, otherwise theater projects without audio will not work.

## Follow-up work
This will be followed up by the Javabuilder-side change to add a `NO_AUDIO` signal.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/javalab/theater/Theater.js [hunk 1]
```diff
@@ -46,14 +46,19 @@ export default class Theater {
         this.openPhotoPrompter(data.detail.prompt);
         break;
       }
+      case TheaterSignalType.NO_AUDIO: {
+        // there is no audio associated with the video, trigger startPlayback so we don't wait for the audio file
+        this.startPlayback();
```
Comment (sanchitmalhotra126) [inline apps/src/javalab/theater/Theater.js:54]: Do we hit any issues calling `play` on an audio element that has no `src`?

**Review by sanchitmalhotra126 — APPROVED** [review]
Nice! 

Re: this comment -
> In testing this I found that Safari also seems to have issues with how we create audio files locally

That's interesting...do you know what the issue is with local files specifically? Is it how we're serving it (local filesystem vs S3)?

**Comment by molly-moen** [comment]
> That's interesting...do you know what the issue is with local files specifically? Is it how we're serving it (local filesystem vs S3)?

@sanchitmalhotra126 I'm not sure. Safari just says the file is not playable.

File: apps/src/javalab/theater/Theater.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/javalab/theater/Theater.js:54]: oh good point. It does throw a console error, I'll clean this up.

**Review by sanchitmalhotra126 — APPROVED** [review]

**Pull request merged**